### PR TITLE
Adding documentation for dataloader.

### PR DESCRIPTION
**Pull request by kartikdutt18**
I have updated readme a little bit to show use of data loaders and pre-processor functions. I have left utility functions out for now. Let me know if I should include them.
One other is I have copied the Contribution.md from mlpack repo.
I think it makes sense to have a wiki page for data loaders, utility functions and pre-processors. Kindly let me know if the wiki page idea makes sense. Thanks a lot.
Looking forward to your suggestions.
Regards.

### CODE REVIEW

**Comment by kartikdutt18** [comment]
Hey @birm, @zoq,  When you get a chance could you please take a look at the documentation that I added in the wiki [here](https://github.com/kartikdutt18/models/wiki/Dataloader-Tutorial). If this makes sense I can add the same for rest of the functions.

**Comment by KimSangYeon-DGU** [comment]
> I have left utility functions out for now. Let me know if I should include them.

@kartikdutt18 Can you let me know what the utility functions are?

**Comment by kartikdutt18** [comment]
> @kartikdutt18 Can you let me know what the utility functions are?

Hey @KimSangYeon-DGU,

Other than the dataloader we also added :
1. PreProcess Class that preprocess data.
2. Utility class which has the following functions : 

    1. DownloadFile : Downloads Files
    2. CheckCRC32 : Performs Checksum
    3. Remove : Removes files from directory
    4. GetCRC32 : Generates CRC32 for a file.

Kindly let me know what you think about the dataloader documentation if that makes sense, I can add similar documentation for the rest of functions.
Thanks a lot.

File: README.md [hunk 1]
```diff
@@ -8,9 +8,16 @@ helping with the transition from the `examples/` repository, be sure to look for
 comments like this one.  Once the transition is done, we can remove this
 comment (and the others).)_
 
-_(If we have functionality to download datasets and also to download
-pretrained model weights, we should put a comment about that here in the main
-description of the repository.)_
+We provide ability to download datasets as well as pretrained weights using our
+utility functions, by default we asume the server to be mlpack.org.
+To dowload any file from mlpack.org simple use following command.
```
Comment (zoq) [inline README.md:13]:
```suggestion
To dowload any file from mlpack.org simple use the following command.
```

File: README.md [hunk 2]
```diff
@@ -21,7 +28,7 @@ out here!)_
   1. [Introduction](#1-introduction)
   2. [Dependencies](#2-dependencies)
   3. [Building-From-Source](#3-building-from-source)
-  4. [Running Models](#4-running-models)
+  5. [Running Models](#5-running-models)
```
Comment (zoq) [inline README.md:31]: Looks like we have to adjust the listing here.

Comment (kartikdutt18) [reply README.md:31]: Sure, I'll make the changes. Thanks a lot.

**Comment by KimSangYeon-DGU** [comment]
Hi @kartikdutt18, in my opinion, it seems to be better for us to keep `README.md` concise focusing on the main points like you've done :)

Regarding functions in `utils`, we already give explanations in detail, so I think it's okay not to write such the info for the additional utility functions. However, if you think we should handle that as well, please go ahead :)

**Comment by kartikdutt18** [comment]
> Hi @kartikdutt18, in my opinion, it seems to be better for us to keep README.md concise focusing on the main points like you've done :)

Great, I'll just add some more properties regarding the dataloader then.

> (Maybe, it's a little hard to keep updating those docs in the future..., so automating approach from comments would be a workaround..., but that's out of this PR)

That would be really great.

**Comment by KimSangYeon-DGU** [comment]
> Great, I'll just add some more properties regarding the dataloader then.

Ok, thanks!

**Comment by kartikdutt18** [comment]
I have updated the wiki with a few corrections. Kindly let me know what you think. This is ready from my side. If the wiki is good to, I think I can port it to this repo as well.

**Comment by KimSangYeon-DGU** [comment]
Wiki is also nice :)

**Comment by kartikdutt18** [comment]
Great, Let me know when to port it to the main models repo. Thanks a lot for the reviews.

**Comment by KimSangYeon-DGU** [comment]
@kartikdutt18 I think after merging this PR, it's good time to add wiki :)

**Comment by kartikdutt18** [comment]
Great. Will add it then.

**Comment by KimSangYeon-DGU** [comment]
@kartikdutt18 Can you look through **Style Checks**?

**Comment by kartikdutt18** [comment]
Done, The style failure was related to files that were stored here earlier. I have done a rebase to fix that :)

File: README.md [hunk 3]
```diff
@@ -72,11 +80,88 @@ building with 4 cores can be done with the following command:
...
+```cpp
+// Use the dataloader for training.
+model.Train(dataloader.TrainFeatures(), dataloader.TrainLabels());
+ 
+// Use the dataloader for prediction.
+model.Predict(dataloader.TestFeatures(), dataloader.TestLabels());
+```
+
+Currently supported datasets are mentioned below :
+((I think we should add a table here with various columns and details.))
```
Comment (birm) [inline README.md:114]: +1
Do you think we should do that now, or in another PR?

Comment (kartikdutt18) [reply README.md:114]: I can do that now. I'll push a commit in couple of minutes.

Comment (kartikdutt18) [reply README.md:114]: I think the columns would be name, type, details. Does this make sense?

Comment (kartikdutt18) [reply README.md]: Done, You could take a look at the table [here](https://github.com/kartikdutt18/models/blob/Dataloader-documentation/README.md).

**Comment by KimSangYeon-DGU** [comment]
@kartikdutt18 Looks good to me. Thanks!

**Comment by kartikdutt18** [comment]
Awesome, Thanks a lot @KimSangYeon-DGU, @birm, @zoq for all the reviews and suggestions.

**Pull request merged**